packages/medusa/src/api-v2/admin/inventory-items/route.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../types/routing"
+import {
+  ContainerRegistrationKeys,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+
+// List inventory-items
+export const GET = async (
+  req: AuthenticatedMedusaRequest,
+  res: MedusaResponse
+) => {
+  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+
+  const query = remoteQueryObjectFromString({
+    entryPoint: "inventory_items",
+    variables: {
+      filters: req.filterableFields,
+      order: req.listConfig.order,
+      skip: req.listConfig.skip,
+      take: req.listConfig.take,
+    },
+    fields: [...(req.listConfig.select as string[])],
+  })
+
+  const { rows: items, metadata } = await remoteQuery({
+    ...query,
+  })
+
+  const inventory_items = items.map((i) => {
+    const [stocked_quantity, reserved_quantity] = i.location_levels.reduce(
```

Comment:
sugg: might be good to check the existence of location_levels or safely traverse here since you can break it here if you send "expand=" without it.

Reply:
changed it up to calculate in the database instead